storage/tree/node_id2.go

```diff
@@ -37,10 +37,28 @@ func NewNodeID2(path string, bits uint) NodeID2 {
 	} else if mx := uint(len(path)) * 8; bits > mx {
 		panic(fmt.Sprintf("NewNodeID2: bits %d > %d", bits, mx))
 	}
-	bytes, tail, mask := split(bits)
-	last := path[bytes] & mask
+	bytes, tailBits := split(bits)
 	// Note: Getting the substring is cheap because strings are immutable in Go.
-	return NodeID2{path: path[:bytes], last: last, bits: tail}
+	return newMaskedNodeID2(path[:bytes], path[bytes], tailBits)
```

Comment:
What are the semantics of the unset bits of `last`.  The comment above makes it sound like they should be unset, but here they look set.

Reply:
Does `path[bytes]` really auto convert from string to byte??

Reply:
> What are the semantics of the unset bits of `last`. The comment above makes it sound like they should be unset, but here they look set.

The bits are unset in order to have a canonical representation of IDs. This helps using `NodeID2` as a Golang map key, for instance, so that 2 IDs constructed independently can compare equal regardless of the way they were created.

> Does `path[bytes]` really auto convert from string to byte??

`path[bytes]` returns a byte straight away. Golang strings contain bytes.

Reply:
The `last` byte is masked in `newMaskedNodeID2`.